### PR TITLE
Balder/optimize bool executors rebased

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR
Only the  3 last PR's are new compared to #12322 
Best viewed individually I think (the last 3)

### CODE REVIEW

File: searchlib/src/tests/features/prod_features.cpp [hunk 1]
```diff
@@ -410,8 +411,8 @@ Test::setupForAttributeTest(FtFeatureTest &ft, bool setup_env)
             .addField(FieldType::ATTRIBUTE, CollectionType::SINGLE, "slong")
             .addField(FieldType::ATTRIBUTE, CollectionType::SINGLE, "sdouble")
             .addField(FieldType::ATTRIBUTE, CollectionType::SINGLE, "sbyte")
-            .addField(FieldType::ATTRIBUTE, CollectionType::SINGLE, "sbool")
-            .addField(FieldType::ATTRIBUTE, CollectionType::SINGLE, "sebool");
+            .addField(FieldType::ATTRIBUTE, CollectionType::SINGLE, DataType::BOOL,"sbool")
```
Comment (havardpe) [inline searchlib/src/tests/features/prod_features.cpp]: in the future: maybe we need to enforce passing the data type

File: streamingvisitors/src/vespa/searchvisitor/searchvisitor.cpp [hunk 2]
```diff
@@ -99,11 +99,11 @@ createAttribute(const vespalib::string & name, const document::FieldValue & fv)
...
 
     if (fv.inherits(document::ByteFieldValue::classId) || fv.inherits(document::IntFieldValue::classId) || fv.inherits(document::LongFieldValue::classId)) {
-        attr.reset(new search::SingleIntegerExtAttribute(name));
+        return std::make_shared<search::SingleIntegerExtAttribute>(name);
     } else if (fv.inherits(document::DoubleFieldValue::classId) || fv.inherits(document::FloatFieldValue::classId)) {
-        attr.reset(new search::SingleFloatExtAttribute(name));
+        return std::make_shared<search::SingleFloatExtAttribute>(name);
     } else if (fv.inherits(document::StringFieldValue::classId)) {
-        attr.reset(new search::SingleStringExtAttribute(name));
+        return std::make_shared<search::SingleStringExtAttribute>(name);
```
Comment (havardpe) [inline streamingvisitors/src/vespa/searchvisitor/searchvisitor.cpp]: should remove attr variable as well

File: streamingvisitors/src/vespa/searchvisitor/searchvisitor.cpp [hunk 3]
```diff
@@ -240,9 +240,9 @@ void SearchVisitor::init(const Parameters & params)
         LOG(debug, "QFLAG_DUMP_FEATURES: %s", _rankController.getDumpFeatures() ? "true" : "false");
     }
 
-    if (params.lookup("rankproperties", valueRef) && valueRef.size() > 0) {
+    if (params.lookup("rankproperties", valueRef) && ! valueRef.empty()) {
         LOG(spam, "Received rank properties of %zd bytes", valueRef.size());
-        uint32_t len = static_cast<uint32_t>(valueRef.size());
+        uint32_t len = uint32_t(valueRef.size());
```
Comment (havardpe) [inline streamingvisitors/src/vespa/searchvisitor/searchvisitor.cpp]: could just construct len directly from valueRef.size() instead of using a temporary

**Review by havardpe — APPROVED** [review]

File: searchlib/src/tests/features/prod_features.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/tests/features/prod_features.cpp]: Agree.

File: streamingvisitors/src/vespa/searchvisitor/searchvisitor.cpp (hunk 2 above)
Comment (baldersheim) [reply streamingvisitors/src/vespa/searchvisitor/searchvisitor.cpp]: Fixed

File: streamingvisitors/src/vespa/searchvisitor/searchvisitor.cpp (hunk 3 above)
Comment (baldersheim) [reply streamingvisitors/src/vespa/searchvisitor/searchvisitor.cpp]: Fixed

**Pull request merged**